packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -0,0 +1,329 @@
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionRequest {
+    requestType: OriginalRequestType.QueryMessageExtensionRequest;
+    commandId: string;
+    parameters?: {
+      name?: string;
+      value?: string;
+    }[];
+    queryOptions?: {
+      count: number;
+      skip: number;
+    };
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  interface IActionExecuteInvokeRequest {
+    requestType: OriginalRequestType.ActionExecuteInvokeRequest;
+    type: string; // "invoke"
+    id: string; // "action id"
+    verb: string; // "action"
+    data: Record<string, unknown>; //object; // {}
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum OriginalRequestType {
+    ActionExecuteInvokeRequest = 'ActionExecuteInvokeRequest',
+    QueryMessageExtensionRequest = 'QueryMessageExtensionRequest',
+  }
+  /*********** END REQUEST TYPE ************/
+
+  /*********** BEGIN RESPONSE TYPE ************/
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type IInvokeResponse = IQueryMessageExtensionResponse | IActionExecuteResponse;
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export enum InvokeResponseType {
+    ActionExecuteInvokeResponse = 'ActionExecuteInvokeResponse',
+    QueryMessageExtensionResponse = 'QueryMessageExtensionResponse',
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IQueryMessageExtensionResponse {
+    responseType: InvokeResponseType.QueryMessageExtensionResponse;
+    composeExtension?: ComposeExtensionResponse;
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export interface IActionExecuteResponse {
+    responseType: InvokeResponseType.ActionExecuteInvokeResponse;
+    value: Record<string, unknown>;
+    signature?: string;
+    statusCode: number;
+    type: string;
+  }
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type ComposeExtensionResponse = {
+    attachmentLayout: AttachmentLayout;
+    type: ComposeResultTypes;
+    attachments: QueryMessageExtensionAttachment[];
+    suggestedActions?: QueryMessageExtensionSuggestedActions;
+    text?: string;
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type QueryMessageExtensionSuggestedActions = {
+    actions?: Actions[];
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type Actions = {
+    type: string;
+    title: string;
+    value: string;
+  };
+
+  /**
+   * @hidden
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export type QueryMessageExtensionCard = {
+    contentType: string;
```

Comment:
Could you @jekloudaMSFT @erinha double check here because I think there is a discrepancy (on parameters inside of `QueryMessageExtensionCard`) between code and doc. Not sure you two had an in-person sync but I was not there so I didn't catch up the latest update.

Reply:
Good question. Doc and host SDK have been updated. Should remove `name`, `thumbnailUrl` and `contentUrl`. FYI @ashishsingh2105 , this will match the host SDK side. Also please note that `content` is of type `Record<string, unknown`. (Doc has recently been updated to match.)